waspc/src/Wasp/Generator/ServerGenerator/EmailG.hs

```diff
@@ -0,0 +1,143 @@
+module Wasp.Generator.ServerGenerator.EmailG where
+
+import Data.Aeson (object, (.=))
+import qualified Data.Aeson as Aeson
+import Data.Maybe (isJust)
+import StrongPath (File', Path', Rel, relfile, (</>))
+import Wasp.AppSpec (AppSpec)
+import qualified Wasp.AppSpec.App as AS.App
+import qualified Wasp.AppSpec.App.Dependency as AS.Dependency
+import Wasp.AppSpec.App.EmailSender (EmailSender)
+import qualified Wasp.AppSpec.App.EmailSender as AS.EmailSender
+import Wasp.AppSpec.Valid (getApp)
+import Wasp.Generator.FileDraft (FileDraft)
+import Wasp.Generator.Monad (Generator)
+import qualified Wasp.Generator.ServerGenerator.Common as C
+import qualified Wasp.SemanticVersion as SV
+import Wasp.Util ((<++>))
+
+genEmail :: AppSpec -> Generator [FileDraft]
+genEmail spec = case maybeEmailSender of
+  Just emailSender ->
+    sequence
+      [ genIndex emailSender
+      ]
+      <++> genCore emailSender
+  Nothing -> return []
+  where
+    maybeEmailSender = AS.App.emailSender $ snd $ getApp spec
+
+genIndex :: EmailSender -> Generator FileDraft
+genIndex email = return $ C.mkTmplFdWithDstAndData srcPath dstPath (Just tmplData)
```

Comment:
I thought we have a `mkTmpl` function that assumes same location in templates and destination -> if we do, maybe you can use that one, so you don't have to repeat it twice in the `where`?

Reply:
Forgot about that, thanks 👍

Reply:
You haven't implemented this yet, right? Or maybe I misremembered and there is no such function?